fixed misaligned elements in generated refunds page issue

SpaceSurfer1 opened the pull request:
Fixed misaligned elements in generated refunds page issue.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes https://github.com/woocommerce/woocommerce/issues/45213 .

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to WP Admin > Pages. If your site already has a "Refund and Returns Policy" page, delete it.
2. Using wp-cli, run `wp shell`. Then in the shell, run `WC_Install::create_pages()`. This will regenerate the Refund page with the updated content from this PR.
3. Back on WP Admin > Pages, click the link to preview the "Refund and Returns Policy" page. The "Late or missing refunds" and "Sale items" headings should now be aligned correctly.

<!-- End testing instructions -->

### Changelog entry

Fixed misaligned elements in generated refunds page issue, updated tags.

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Fix alignment issues in the generated content of the Refunds page

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

SpaceSurfer1 committed:
fixed misaligned elements in generated refunds page issue

Fixed misaligned elements in generated refunds page issue.

coreymckrill commented:
@SpaceSurfer1 thanks for the PR! In my testing, looks like this does indeed fix the issue. I wonder if it would be even better, from an HTML semantics perspective, if those two lines were changed to `h3` tags instead of paragraphs. What do you think?

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

SpaceSurfer1 commented:
Hi @coreymckrill, Thanks for reviewing👍. I just tested with `h3` tag and compared with present one, following is the screenshot -

![Screenshot 2024-03-05 at 2 09 19 AM](https://github.com/woocommerce/woocommerce/assets/99202999/315a5843-7623-4919-bdc8-0d8f369721e9)

The font and size is changed while using `h3`. `<p><b>` keeps it same as before. Any suggestions?

coreymckrill commented:
🤔 I guess it depends on which theme you're using. It looks like you're testing on Twenty Twenty Four, which is what I'm using too. With the h3, it is definitely bigger than it was before, but it's still smaller than the h2 heading that it belongs to:

![refunds](https://github.com/woocommerce/woocommerce/assets/916023/30e11b57-fb05-4a2a-8417-8e19c8955fd3)

I'll check and see if anyone else has thoughts on this...

SpaceSurfer1 commented:
@coreymckrill , Sure....👍

coreymckrill commented:
@SpaceSurfer1 given that this change doesn't affect current stores, only new ones (since the content is only generated once when WooCommerce is first installed), I'd say let's go ahead and improve the semantics and switch to `h3`s.

This actually brings up a few other issues with the generated content that you could also address in this PR if you want to (also fine to just keep it focused on the two lines):

* "Overview" is currently an `h3`, but it should probably be an `h2` to match the other heading levels
* All of the headings are wrapped in the wrong block "grammar". Meaning they are wrapped in `<!-- wp:paragraph -->` block tags, but they should have `<!-- wp:heading -->` instead.
* The heading tags should have a `class="wp-block-heading"` attribute, since that's what you get when you insert a heading block into the editor manually.

So the "Overview" heading, for example, should look like this:

```
<!-- wp:heading -->
<h2 class="wp-block-heading">Overview</h2>
<!-- /wp:heading -->
```

If you'd rather not take on the extra changes, we can spin up another PR later.

SpaceSurfer1 commented:
Hi @coreymckrill , I'll address those additional changes in this PR only. 👍

SpaceSurfer1 committed:
Updated tags and added class attribute for headings

Updated tags to <h3>, updated block tags and added class attribute for headings

SpaceSurfer1 committed:
Merge branch 'fix/misaligned-elements-in-generated-refunds-page' of https://github.com/SpaceSurfer1/woocommerce into fix/misaligned-elements-in-generated-refunds-page

SpaceSurfer1 commented:
Hi @coreymckrill , I have updated the tags and added class attribute👍 . Can you please review it? Thanks.

SpaceSurfer1 committed:
Merge branch 'woocommerce:trunk' into fix/misaligned-elements-in-generated-refunds-page

coreymckrill commented on plugins/woocommerce/includes/class-wc-install.php:
```diff
@@ -2284,13 +2284,13 @@ private static function set_paypal_standard_load_eligibility() {
-<!-- wp:paragraph -->
-<p><b>This is a sample page.</b></p>
-<!-- /wp:paragraph -->
+<!-- wp:heading -->
+<h3 class="wp-block-heading">This is a sample page.</h3>
+<!-- /wp:heading -->
```
```suggestion
<!-- wp:paragraph -->
<p><b>This is a sample page.</b></p>
<!-- /wp:paragraph -->
```

Even though this one is also bolded, I don't think it's intended to be a header, but rather just a note to the site owner, so I think this should stay as a paragraph.

coreymckrill committed:
Update plugins/woocommerce/includes/class-wc-install.php

coreymckrill reviewed (approved):
👍 All looks good now
